protocol/protocol/node-sync/traits-and-abstractions.md

```diff
@@ -0,0 +1,122 @@
+- Uses `Syncer` for sync coordination, emits `ConsensusEvent` for system state
+
+**`ConsensusProxy`**
+
+- Thread-safe, cloneable interface to consensus functionality
+- Provides transaction sending, state queries, block resolution
+
+**`Syncer`**
+
+- Manages macro sync → live sync transitions and peer compatibility
+- Bridges between different sync phases and handles peer state tracking
+- Contains `macro_sync` and `live_sync` trait objects for different strategies
+
+**`SyncerProxy`**
+
+- Enum wrapper that abstracts over different node sync combinations: History/Full/Light/Pico
+- Each variant contains a `Syncer` with specific `MacroSync` and `LiveSync` implementations
+
+**`SyncQueue`**
+
+- Generic request coordinator with peer rotation, retry logic, and response verification
+- Used by all sync implementations for network requests
+- Maintains ordered response processing with custom verification callbacks
+
+**Request Components**
+
+- `BlockRequestComponent` - Missing block requests
+- `ChunkRequestComponent` - State chunk requests
+- `DiffRequestComponent` - Trie diff requests
+
+### Blockchain Services
+
+| Component | Purpose | Key Features |
+|-----------|---------|--------------|
+| `RemoteDataStore<N>` | Remote staking contract queries | Merkle proof verification, validator/staker data |
+| `RemoteEventDispatcher<N>` | Address-based event routing | Client subscriptions, event notifications |
+| `BlsCache` | BLS key optimization | LRU cache, validator voting key storage |
+| `HeadRequests<N>` | Consensus peer analysis | Head state tracking, missing block detection |
+
+## Component Interactions
+
+### Event-Driven Communication
+
+Components coordinate through event streams rather than direct method calls. The consensus system emits `ConsensusEvent::Established` and `ConsensusEvent::Lost` to signal state changes, while sync components emit `LiveSyncEvent` for block processing and peer status.
+
+### Request Coordination Pattern
+
+All sync strategies use `SyncQueue` for reliable peer coordination. When data is needed, sync implementations coordinate requests across multiple peers with automatic retry, peer rotation, and response verification.
+
+### Proxy Pattern Architecture
+
+`ConsensusProxy` provides thread-safe access via channels, `SyncerProxy` is an enum wrapper for different sync strategies (History/Full/Light/Pico).
+
+### Stream-Based Processing
+
+Key coordination components implement `Stream` trait for async coordination: `MacroSync`, `LiveSync`, `LiveSyncQueue`, and `LiveSyncer`.
+
+### Generic Queue Architecture
+
+Same live sync coordination logic works with different processing strategies: `LiveSyncQueue` trait lets `BlockQueue`, `StateQueue`, and `DiffQueue` plug into `LiveSyncer`.
+
+### Fallback Mechanisms
+
+`EitherSyncer` enables `PicoMacroSync` → `LightMacroSync` transitions when conflicts are detected.
+
+## TL;DR
+
+1. **`MacroSync`** → Node-specific strategies to reach current network state
```

Comment:
Suggestion:
1. **`MacroSync`** → Node-specific strategies to reach the latest macro block